src/pages/billing/Project.js

```diff
@@ -153,32 +210,36 @@ const ProjectDetail = ({ project, billingAccounts, authorizeAndLoadAccounts }) =
 
   const adminCanEdit = _.filter(({ roles }) => _.includes(billingRoles.owner, roles), projectUsers).length > 1
 
-  const getBillingAccountStatusIcon = workspace => Utils.cond(
-    [_.isEmpty(billingProject.workspacesWithIncorrectBillingAccount), () => null],
-    [billingProject.billingAccount === workspace.billingAccount, () => 'success-standard'],
-    ['billingAccountErrorMessage' in workspace, () => 'error-standard'],
-    [Utils.DEFAULT, () => 'loadingSpinner']
-  )
+  const workspacesInProject = useMemo(() => _.flow(
+    _.map('workspace'),
+    _.filter({ namespace: billingProject.projectName })
+  )(workspaces), [billingProject, workspaces])
+
+  const groups = groupByBillingAccountStatus(billingProject, workspacesInProject)
+  const billingAccountsOutOfDate = !_.every(_.isEmpty, [groups.error, groups.updating])
+  const getBillingAccountStatus = workspace => _.findKey(g => g.has(workspace), groups)
```

Comment:
I might rename `g` to `ws` ("workspaces set") or `set` to make it clear that the members of `groups` are sets.

Reply:
Is there a generic way of telling if a collection contains some element? The fact I'm using a set is an implementation detail (for `O(1)` lookup) - really this code should work for any collection.

Reply:
Take a quick look at the Lodash documentation for `_.some`. It has some magic that may work for your use case here. That said, I think your usage of `Set` here clarifies the intent and may be the best choice.